app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -194,31 +189,25 @@ class ParkingViewModel(
         Point.fromLngLat(
             maxOf(startPos.longitude(), endPos.longitude()),
             maxOf(startPos.latitude(), endPos.latitude()))
-
     // Get all tiles that are in the rectangle
-    tilesToDisplay = Tile.getAllTilesInRectangle(bottomLeft, topRight)
-    // Used to keep track of when the last request has finished.
+    val tilesToDisplay = TileUtils.getAllTilesInRectangle(bottomLeft, topRight)
     var nbRequestLeft = tilesToDisplay.size
 
     tilesToDisplay.forEach { tile ->
 
-      // Get parkings from memory cache
-      if (tileCache.containsKey(tile.uid)) {
-        _rectParkings.value += tileCache[tile.uid]!!.parkings
+      //  Cache hit
+      if (tilesToParking.containsKey(tile)) {
+        _rectParkings.value += tilesToParking[tile]!!
         updateClosestParkings(--nbRequestLeft)
 
-        // Get parkings from repository
+        // Cache miss
       } else {
-        // Cache tile
-        tileCache[tile.uid] = tile
-        Log.d("ParkingViewModel", "Requesting parkings for tile ${tile.uid}")
-        parkingRepository.getParkingsBetween(
-            tile.bottomLeft,
-            tile.topRight,
+        tilesToParking[tile] = emptyList()
```

Comment:
Why is this here? This is "false" branching of the if condition so that means that tilesToParking[tile] is already an empty list, no?

Reply:
Since requests to Firestore are asynchronous, if we have a second call to `getParkingsInRect`, it will make a second request to firestore if the first one did not complete before. This way the empty list marks this tile as "already" queried

Reply:
Good implementation then!